File: althex/determine_case.py
# Redefine the functions after the reset.

althex_chars = set('GHJKLMNOPQRSTUVWXYZghjklmnopqrstuvwxyz')
stdhex_chars = set('ABCDEFabcdef')

def is_upper_althex(s):
    filtered_chars = set(filter(althex_chars.__contains__, s))
    return bool(filtered_chars) and "".join(filtered_chars).isupper()


def is_lower_althex(s):
    filtered_chars = set(filter(althex_chars.__contains__, s))
    return bool(filtered_chars) and "".join(filtered_chars).islower()


def is_upper_stdhex(s):
    filtered_chars = set(filter(stdhex_chars.__contains__, s))
    return bool(filtered_chars) and "".join(filtered_chars).isupper()


def is_lower_stdhex(s):
    stdhex_chars = set('abcdef')
    filtered_chars = set(filter(stdhex_chars.__contains__, s))
    return bool(filtered_chars) and "".join(filtered_chars).islower()

File: glitch-projects/ehime-u/althex/althex-js/determine_case.js
const althexChars = new Set('GHJKLMNOPQRSTUVWXYZghjklmnopqrstuvwxyz');
const stdhexChars = new Set('ABCDEFabcdef');

function isUpperAlthex(s) {
    const filteredChars = [...s].filter(c => althexChars.has(c));
    return filteredChars.length > 0 && filteredChars.every(c => c === c.toUpperCase());
}

function isLowerAlthex(s) {
    const filteredChars = [...s].filter(c => althexChars.has(c));
    return filteredChars.length > 0 && filteredChars.every(c => c === c.toLowerCase());
}

function isUpperStdhex(s) {
    const filteredChars = [...s].filter(c => stdhexChars.has(c));
    return filteredChars.length > 0 && filteredChars.every(c => c === c.toUpperCase());
}

function isLowerStdhex(s) {
    const filteredChars = [...s].filter(c => stdhexChars.has(c));
    return filteredChars.length > 0 && filteredChars.every(c => c === c.toLowerCase());
}
